tests/test_E3_indicator_calculation.py

```diff
@@ -626,19 +627,20 @@ def test_add_onsite_energy_matching():
 def test_add_degree_of_autonomy():
     """ """
 
-    total_generation = 50
+    total_consumption_from_grid = 50
     total_demand = 100
     dict_values_DA = {
         KPI: {
             KPI_SCALARS_DICT: {
-                TOTAL_GENERATION_IN_LES: total_generation,
+                TOTAL_CONSUMPTION_FROM_PROVIDERS
+                + SUFFIX_ELECTRICITY_EQUIVALENT: total_consumption_from_grid,
                 TOTAL_DEMAND + SUFFIX_ELECTRICITY_EQUIVALENT: total_demand,
             }
         },
     }
     E3.add_degree_of_autonomy(dict_values_DA)
 
-    degree_of_autonomy = total_generation / total_demand
+    degree_of_autonomy = (total_demand - total_consumption_from_grid) / total_demand
 
     assert (
         dict_values_DA[KPI][KPI_SCALARS_DICT][DEGREE_OF_AUTONOMY] == degree_of_autonomy
```

Comment:
The assertion message here in `test_add_degree_of_autonomy` seems to be wrong (not shown in this view)

Reply:
Improved this